src/Type/SubtractedType.php

```diff
@@ -0,0 +1,313 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PHPStan\Reflection\ClassMemberAccessAnswerer;
+use PHPStan\Reflection\ConstantReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\PropertyReflection;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\Generic\TemplateTypeMap;
+use PHPStan\Type\Generic\TemplateTypeVariance;
+
+class SubtractedType implements Type, CompoundType
+{
+	private $originalType;
+	private $subtractedType;
+
+	private function __construct(Type $originalType, Type $subtractedType)
+	{
+		$this->originalType = $originalType;
+		$this->subtractedType = $subtractedType;
+	}
+
+	/**
+	 * TODO: move/use logic from TypeCombinator here, to create the type
+	 * @see TypeCombinator::remove()
+	 * The goal should be to normalize the type if possible (subtract a class from a union type of classes etc)
+	 * Only if that is not possible use this type, e.g. when subtracting a constant string from the general string type
+	 */
+	public static function create(Type $originalType, Type $subtractedType): Type
```

Comment:
See TODO, this type should only be used as a last resort, if we can subtract the types easily e.g. remove one type from a union or intersection then that should be done instead.